### PR TITLE
Frame reordering

### PR DESCRIPTION
**Pull request by fbossen**
Group frames in sets of 4. Encode the far one first and then the other three using a forward and a backward reference.

### CODE REVIEW

File: src/api.rs [hunk 1]
```diff
@@ -54,15 +54,17 @@ impl Config {
...
 }
 
 pub struct Context {
   fi: FrameInvariants,
   seq: Sequence,
   //    timebase: Ratio,
-  frame_q: VecDeque<Option<Arc<Frame>>> //    packet_q: VecDeque<Packet>
+  frame_count: u64,
+  idx: u64,
+  frame_q: VecDeque<(u64, Option<Arc<Frame>>)> //    packet_q: VecDeque<Packet>
```
Comment (lu-zero) [inline src/api.rs]: You might want to use a map probably e.g. [btreemap](https://doc.rust-lang.org/stable/std/collections/struct.BTreeMap.html).

Comment (fbossen) [reply src/api.rs]: You are right. A BTreeMap is much cleaner. Thanks for the tip!

File: src/api.rs [hunk 2]
```diff
@@ -168,9 +168,9 @@ impl Context {
       self.fi.base_q_idx = (self.fi.config.quantizer.min(255 - q_drop) + q_drop) as u8;
 
       let first_ref_frame = LAST_FRAME;
-      let second_ref_frame = ALTREF_FRAME;
+      let second_ref_frame = if idx_in_group == 0 { LAST2_FRAME } else { ALTREF_FRAME };
 
-      self.fi.primary_ref_frame = (first_ref_frame - LAST_FRAME) as u32;
+      self.fi.primary_ref_frame = PRIMARY_REF_NONE; //(first_ref_frame - LAST_FRAME) as u32;
```
Comment (tdaede) [inline src/api.rs]: From my limited experience in libaom, you generally want to set this to whatever reference frame was at the same pyramid level (and none ended up better than a random last frame, as you also seem to have found out)

**Comment by tdaede** [comment]
LGTM, but needs a rebase.

**Pull request merged**